feat(website): new documentation (#1048)
algolia/docsearch: packages/website/docs/required-configuration.mdx

---
title: Required configuration
---

This section, gives you the best practices to optimize our crawl. Adopting this following specification is required to let our crawler build the best experience from your website. You will need to update your website and follow these rules.

:::note

> If your website is generated, thanks to one of [our supported tool][1], you do not need to change your website as it is already compliant with our requirements.

Reviewer: ```suggestion
If your website is generated, thanks to one of [our supported tools][1], you do not need to change your website as it is already compliant with our requirements.
```

I think you have more than one supported tool?

If you are using one of [our integrations][13], please see [the templates page][11].

<details><summary>docsearch-default.js</summary>
<div>

```js
new Crawler({
  appId: 'YOUR_APP_ID',
  apiKey: 'YOUR_API_KEY',
  startUrls: ['https://YOUR_START_URL.io/'],
  sitemaps: ['https://YOUR_START_URL.io/sitemap.xml'],
  actions: [
    {
      indexName: 'YOUR_INDEX_NAME',
      pathsToMatch: ['https://YOUR_START_URL.io/**'],
      recordExtractor: ({ helpers }) => {
        return helpers.docsearch({
          recordProps: {
            lvl0: {
              selectors: '',
              defaultValue: 'Documentation',
            },
            lvl1: 'header h1',
            lvl2: 'article h2',
            lvl3: 'article h3',
            lvl4: 'article h4',
            lvl5: 'article h5',
            lvl6: 'article h6',
            content: 'article p, article li',
          },
        });
      },
    },
  ],
  initialIndexSettings: {
    YOUR_INDEX_NAME: {
      // These attributes can be used as `facetFilters` in the frontend
      attributesForFaceting: ['type'],
      // Below are all the default settings of your Algolia index in order to

>       // make DocSearch v3 work. Changing them can cause issue with the final

Reviewer: ```suggestion
      // make DocSearch v3 work. Changing them can cause issues with the final
```

      minWordSizefor1Typo: 3,
      minWordSizefor2Typos: 7,
      allowTyposOnNumericTokens: false,
      minProximity: 1,
      ignorePlurals: true,
      advancedSyntax: true,
      attributeCriteriaComputedByMinProximity: true,
      removeWordsIfNoResults: 'allOptional',
      separatorsToIndex: '_',
    },
  },
});
```

</div>
</details>

### Overview of a clear layout

A website implementing these good practises will look simple and crystal clear. It can have this following aspect:

<img
  src="https://docsearch.algolia.com/img/assets/recommended-layout.png"
  alt="Recommended layout for your page"
/>

The main blue element will be your `.DocSearch-content` container. More details in the following guidelines.

### Use the right classes as [`recordProps`][2]

You can add some specific static classes to help us find your content role. These classes can not involve any style changes. These dedicated classes will help us to create a great learn-as-you-type experience from your documentation.

- Add a static class `DocSearch-content` to the main container of your textual content. Most of the time, this tag is a `<main>` or an `<article>` HTML element.

> - Every searchable `lvl` elements outside this main documentation container (for instance in a sidebar) must be `global` selectors. They will be globally picked up and injected to every record built from your page. Be careful, the level value matters and every matching element must have an increasing level along the HTML flow. A level `X` (for `lvlX`) should appear after a level `Y` while `X > Y`.

Reviewer: ```suggestion
- Every searchable `lvl` element outside this main documentation container (for instance in a sidebar) must be a `global` selector. They will be globally picked up and injected to every record built from your page. Be careful, the level value matters and every matching element must have an increasing level along the HTML flow. A level `X` (for `lvlX`) should appear after a level `Y` while `X > Y`.
```

- `lvlX` selectors should use the standard title tags like `h1`, `h2`, `h3`, etc. You can also use static classes. Set a unique `id` or `name` attribute to these elements as detailed below.

> - Every DOM elements matching the `lvlX` selectors must have a unique `id` or `name` attribute. This will help the redirection to directly scroll down to the exact place of the matching elements. These attributes define the right anchor to use.

Reviewer: ```suggestion
- Every DOM element matching the `lvlX` selectors must have a unique `id` or `name` attribute. This will help the redirection to directly scroll down to the exact place of the matching elements. These attributes define the right anchor to use.
```
